modules/Microsoft.Batch/batchAccounts/deploy.bicep

```diff
@@ -90,14 +86,27 @@ param tags object = {}
 @description('Optional. List of allowed authentication modes for the Batch account that can be used to authenticate with the data plane.')
 param allowedAuthenticationModes array = []
 
-@description('Optional. The resource ID of a key vault to reference a customer managed key for encryption from.')
-param cMKKeyVaultResourceId string = ''
+@allowed([
+  'Microsoft.Batch'
+  'Microsoft.KeyVault'
+])
+@description('Optional. Type of the key source.')
+param encryptionKeySource string = 'Microsoft.Batch'
+
+@description('Conditional. Full path to the versioned secret. Required if `encryptionKeySource` is set to `Microsoft.KeyVault` or `poolAllocationMode` is set to `UserSubscription`.')
+param encryptionKeyIdentifier string = ''
+
+@description('Conditional. The resource ID of the Azure key vault associated with the Batch account. Required if `encryptionKeySource` is set to `Microsoft.KeyVault` or `poolAllocationMode` is set to `UserSubscription`.')
+param keyVaultResourceId string = ''
 
-@description('Optional. The name of the customer managed key to use for encryption.')
-param cMKKeyName string = ''
+@description('Conditional. The URL of the Azure key vault associated with the Batch account. Required if `encryptionKeySource` is set to `Microsoft.KeyVault` or `poolAllocationMode` is set to `UserSubscription`.')
+param keyVaultUri string = ''
```

Comment:
I see the CMK logic was changed together with related parameter names. The original intention was to align that for all modules as much as possible, although unfortunately every resource seems to implement this feature differently. I'd suggest if possible to keep the consistency with other modules. Tagging @MrMCake for additional thoughts.